metals/src/main/scala/scala/meta/internal/metals/onTypeRangeFormatters/OnTypeRangeFormatter.scala

```diff
@@ -0,0 +1,36 @@
+package scala.meta.internal.metals.onTypeRangeFormatters
+
+import scala.meta.tokens.Tokens
+
+import org.eclipse.lsp4j.FormattingOptions
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.Range
+import org.eclipse.lsp4j.TextEdit
+
+trait OnTypeRangeFormatter {
```

Comment:
Could we split both into different traits? A method can be both, but sometimes we might want to implement only one method.

Reply:
It was already possible to implement only one method, just by using the default implementation of the correct `contribute`, but I've split them anyway!